src/publish/s3.js

```diff
@@ -3,9 +3,22 @@ const winston = require("winston-color")
 const zlib = require("zlib")
 const config = require("../config")
 
-const S3 = new AWS.S3()
+let conf = {}
 
+// This is the case where using custom s3 api-like services like minio.
+if (config.aws.endpoint) {
+  conf = {
+    accessKeyId: process.env.AWS_ACCESS_KEY_ID,
+    secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
+    endpoint: (config.aws.endpoint ? config.aws.endpoint : ''),
+    s3ForcePathStyle: 'true',
+    signatureVersion: 'v4'
+  }
```

Comment:
@jmhooper What are your thoughts here?

FWIW, I wasn't suggesting removing the ternary operator -- I'd more like to remove the whole conditional around the `conf` block, so that we're executing the same code in either case, but swapping out the endpoint as needed.